### PR TITLE
Backend study conversion

### PR DESCRIPTION
**Pull request by Kris2ffer**
Adds methods to convert from input model to internal model and back.

### CODE REVIEW

File: RestApi/Sepes.RestApi/Model/Pod.cs [hunk 1]
```diff
@@ -35,11 +44,47 @@ public class Pod
...
             ImmutableList<User>.Empty, ImmutableList<DataSet>.Empty, ImmutableList<DataSet>.Empty) {}
 
+        public PodInput ToPodInput()
+        {
+            var inputIncoming = new List<RuleInput>();
+            var inputOutgoing = new List<RuleInput>();
+            var inputUsers = new List<string>();
+            
+            foreach (Rule rule in incoming) {
+                inputIncoming.Add(new RuleInput(){port = rule.port, ip = rule.ip.ToString()});
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Model/Pod.cs]: It would be better if Rule also had a ToRuleInput().

File: RestApi/Sepes.RestApi/Model/PodInput.cs [hunk 2]
```diff
@@ -9,5 +11,25 @@ public class PodInput
...
+
+
+        public Pod ToPod()
+        {
+            var podIncoming = new List<Rule>();
+            var podOutgoing = new List<Rule>();
+            var podUsers = new List<User>();
+            
+            foreach (RuleInput rule in incoming) {
+                podIncoming.Add(new Rule(rule.port, rule.ip));
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Model/PodInput.cs]: Also here have a ToRule() on Rule.

File: RestApi/Sepes.RestApi/Model/Study.cs [hunk 3]
```diff
@@ -36,13 +39,52 @@ public class Study
...
-        public StudyInput ToStudyInput() // will be fully implemented later
+        public StudyInput ToStudyInput()
         {
+            var inputPods = new HashSet<PodInput>();
+            var inputSponsors = new HashSet<string>();
+            var inputSuppliers = new HashSet<string>();
+
+            foreach (Pod pod in pods) {
+                inputPods.Add(pod.ToPodInput());
+            }
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Model/Study.cs:50]: Tip: Can be simplified using Linq.  var inputPods = pods.Select(p => p.ToPodInput()).ToHashSet();
But may be a preference thing of loops vs iteration.

**Review by Unicus-Tom — APPROVED** [review]

**Review by gjersvik — CHANGES_REQUESTED** [review]
Rule and InputRules should have ToMethods. For simpler future changes even if is a one liner.

Most of the code could be simplified greatly by using Linq methods documented here:
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable?view=netcore-3.0 but that is only a tip. Not something wrong with the PR.

**Review by gjersvik — APPROVED** [review]
Fixed main issue. Happy merge.

**Pull request merged**